### DISCUSSION TITLE
youtube video takes 7 seconds to start with yt-dlp

### DISCUSSION THREAD

**Discussion by geextahslex** [mpv-player/mpv#14363] [General]
Hi, I wanted to ask if this is normal that it takes about 7 seconds to start a youtube video using yt-dlp?
Is there a way to speed this up?

Thank you :)

**Comment by guidocella** [comment]
`--script-opts-append=ytdl_hook-try_ytdl_first=yes`

**Reply by geextahslex** [reply]
Where do I put this line? yt-dlp.conf?
Oh I think mpv.conf
So this makes it start in about 5 seconds.

**Reply by guidocella** [reply]
That's just yt-dlp being slow. I get youtube stream URLs with youtube-viewer which is much faster.

**Reply by geextahslex** [reply]
Okay what is youtube-viewer? Does it also work with mpv or is it something different?

**Reply by guidocella** [reply]
https://github.com/trizen/youtube-viewer

**Reply by geextahslex** [reply]
is it `script-opts-append=` or just `script-opts=` ?

**Reply by guidocella** [reply]
`--script-opts` overwrites previously defined script options.

**Comment by hooke007** [comment]
dup https://github.com/mpv-player/mpv/discussions/14361#discussioncomment-9780325

**Comment by kasper93** [comment]
You should move this discussion to ytdl version that you use. They might have good suggestions how to configure it for faster extraction. I know that yt-dlp for example extracts formats from various clients like web, android, ios and whatever more. It takes time to extract all possible formats.

**Reply by geextahslex** [reply]
I would like to try that but it seems they disabled the discussion section ^^'
There is a discord link, I try it there

**Reply by geextahslex** [reply]
So the suggested solution was to add `--extractor-args "youtube:player_skip=webpage,configs,js;player_client=ios"` to yt-dlp.conf
but this downgrades opus to AAC and you lose the ability to skip forward/seek

**Reply by kasper93** [reply]
win some, lose some. Some player types will give you different formats. Best is to minimize to what you need, but the tricky part is to track the youtube changes to do that.
